stacks_queues/queue.py

```diff
@@ -23,40 +23,78 @@ def enqueue(self, element):
             In the store are occupied
             returns None
         """
-        pass
+        # if self.buffer_size > 20:
+        #     raise QueueFullException()
+        is_added = False
+        for i in range(len(self.store)):
+            if self.store[i] is None:
+                self.store[i] = element
+                is_added = True
+                break
+        if not is_added:
+            raise QueueFullException
+        
 
     def dequeue(self):
         """ Removes an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
             returns None
         """
-        pass
+        if self.empty():
+            raise QueueEmptyException
+        val = self.store[0]
+        self.store.pop(0)
+        self.store.append(None)
+        return val
+        
+        
+
 
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
+        return self.store[0]
         
 
     def size(self):
         """ Returns the number of elements in
             The Queue
         """
-        pass
+        count = 0
+        for element in self.store:
+            if element is not None:
+                count += 1
+        return count
 
     def empty(self):
         """ Returns True if the Queue is empty
             And False otherwise.
         """
-        pass
+        count = 0
+        for element in self.store:
+            if element is not None:
+                count += 1
+        
+        if count == 0:
+            return True
+        else:
+            return False
 
     def __str__(self):
```

Comment:
⚠️ This passes the tests, but will need to be reworked when you implement a circular buffer.